installers/npm/binary.js

```diff
@@ -136,6 +140,61 @@ const install = () => {
   }
 };
 
+/**
+ * Selects the correct env based on the protocol of the request URL.
+ * @param requestURL URL from binary object
+ * @returns proxy env string
+ */
+const getProxyEnv = (requestURL) => {
+  const protocol = requestURL.match(/^\w+/)[0];
+  const noProxy = process.env.NO_PROXY || process.env.no_proxy || "";
+
+  // if the noProxy is a wildcard then return null
+
+  if (noProxy === "*") {
+    return null;
+  }
+
```

Comment:
not sure how this will help. this is under assumption that no_proxy is *
what if
no-proxy=.github.com,.foo.com,.bar.com